File: app/src/androidTest/java/com/example/task2/TestBenchmarkComponent.java
package com.example.task2;

import com.example.task2.models.BenchmarkComponent;
import com.example.task2.ui.benchmark.BenchmarksViewModelFactory;

import dagger.Component;

@Component(modules = {TestBenchmarkModule.class})
public interface TestBenchmarkComponent extends BenchmarkComponent {
    void inject(BenchmarksViewModelFactory factory);
}

File: app/src/androidTest/java/com/example/task2/models/TestBenchmarkComponent.kt
package com.example.task2.models

import com.example.task2.ui.benchmark.BenchmarksViewModelFactory
import dagger.Component

@Component(modules = [TestBenchmarkModule::class])
interface TestBenchmarkComponent : BenchmarkComponent {
    override fun inject(factory: BenchmarksViewModelFactory?)
}
